### PR TITLE
Add Retry button if a BTCPay Vault operation fails

### LINKED ISSUE #1642: Make Vault Page a more informative and add retry
**Is your feature request related to a problem? Please describe.**
The vault page which appears once you sign with a hardware wallet needs more information that will provide a brief info on what's btcpay server vault and have a retry button since very often in case of an error user is unaware that they have to refresh the page to try again.

**Describe the solution you'd like**

- [ ] Add retry button in case of an error?
- [ ] Add a bit more information on the vault (1 sentence with a link to vault page in our docs)

**Additional context**
Capture

### PR DESCRIPTION
**Pull request by NicolasDorier**
Fix #1642 

image

image

### CODE REVIEW

**Review by Kukks — CHANGES_REQUESTED** [review]

File: BTCPayServer/Views/Stores/AddDerivationSchemes_HardwareWalletDialogs.cshtml [hunk 1]
```diff
@@ -95,8 +95,9 @@
                 </div>
             </div>
             <div class="modal-footer">
-                <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+                <button class="btn btn-primary vault-retry" style="display:none;" onclick="return false;">Retry</button>
```
[the remark this answers is not in the document]
Comment (NicolasDorier) [reply BTCPayServer/Views/Stores/AddDerivationSchemes_HardwareWalletDialogs.cshtml:98]: If I remove the onclick, the page make a POST

File: BTCPayServer/wwwroot/js/vaultbridge.ui.js [hunk 2]
```diff
@@ -73,6 +73,15 @@ var vaultui = (function () {
         this.psbt = null;
 
         this.xpub = null;
+
+        this.retryShowing = false;
+
+        function showRetry() {
+            var button = $(".vault-retry");
+            self.retryShowing = true;
```
[the remark this answers is not in the document]
Comment (NicolasDorier) [reply BTCPayServer/wwwroot/js/vaultbridge.ui.js:81]: it is

File: BTCPayServer/Views/Wallets/WalletSendVault.cshtml [hunk 3]
```diff
@@ -26,6 +26,7 @@
             <partial name="SigningContext" for="SigningContext" />
         </form>
         <div id="vaultPlaceholder"></div>
+        <button class="btn btn-primary vault-retry" style="display:none;" onclick="return false;">Retry</button>
```
[the remark this answers is not in the document]
Comment (NicolasDorier) [reply BTCPayServer/Views/Wallets/WalletSendVault.cshtml:29]: I don't think so. ping @pavlenex

File: BTCPayServer/wwwroot/js/vaultbridge.ui.js [hunk 4]
```diff
@@ -149,7 +159,28 @@ var vaultui = (function () {
...
+                    content.html('');
+                    ///////////////////
+                    button.hide();
+                    self.retryShowing = false;
+                    resolve(true);
+                });
+            });
+        };
+
         this.ensureConnectedToBackend = async function () {
```
[the remark this answers is not in the document]
Comment (NicolasDorier) [reply BTCPayServer/wwwroot/js/vaultbridge.ui.js:180]: needed, I use await in the body of the method

File: BTCPayServer/Views/Stores/AddDerivationSchemes_HardwareWalletDialogs.cshtml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kukks) [reply BTCPayServer/Views/Stores/AddDerivationSchemes_HardwareWalletDialogs.cshtml:98]: not if you set the type to button instead of the default (submit)

**Pull request merged**